### DISCUSSION TITLE
Generate internationalized metadata for static pages

### DISCUSSION THREAD

**Discussion by m4rvelous252** [amannn/next-intl#1183] [Q&A] [answered]
Hi guys, just asking around to see if anyone has implemented a next project with some statically generated pages and internationalized metadata info.

I'm building a site with next-intl and sanity as cms, futher context can be found [here](https://nextjs-forum.com/post/1259742954369712138).

Right now I'm struggling to find a way that uses `generateMetadata()` without turning the page into a server rendered page on request

**Comment by amannn** [comment, marked as answer]
Hey @m4rvelous252, if you implement metadata as shown in [the corresponding docs](https://next-intl-docs.vercel.app/docs/environments/actions-metadata-route-handlers#metadata-api) then SSG should work correctly. The key is providing the `locale` to functions like `getTranslations` (see also the green callout at the bottom of the section).

Hope this helps!

**Reply by sfwnisme** [reply]
in my project SSG works fine, meanwhile I use `getTranslations("...")` instead of `getTranslations({locale, namespace: ...})`.
the build results indicates that the routes generated as SSG, as attached.

would the issue appears on another side or way, or it may happen on the future or specific phase, so I have to use the recommended way in the docs?

<img width="923" height="941" alt="image" src="https://github.com/user-attachments/assets/c4b6899f-d45b-4c94-bb2b-a6899d56ef67" />

**Comment by m4rvelous252** [comment]
Thanks for the tips, I just found out the page was being server rendered because of a revalidation setup in the request for the metadata not the use of `generateMetadata()` itself
